File: CSES_problems/Introdutory/NumberSpiral.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl "\n"
#define fastio cin.tie(nullptr), ios_base::sync_with_stdio(false)
typedef long long ll;

int main(){
    fastio;
    int t; cin >> t;
    while (t--)
    {
        ll x,y; cin >> y >> x;
        ll level = max(x,y);
        if(level%2){
            if(x == level){
                cout << (level*level) - (y-1) << endl;
            }
            else{
                cout << ((level*level) - pow(2,level-1)) + (x-1) << endl;
            }
        }
        else{
            if(x == level){
                cout << ((level*level) - pow(2,level-1)) + (y-1) << endl;
            }
            else{
                cout << (level*level) - (x-1) << endl;
            }

        }

    }
    

}

File: EXERCICES/CSES_problems/Introdutory/NumberSpiral.py
t = int(input())

while(t):
    string = input().split()
    y = int(string[0])
    x = int(string[1])
    level = max(x,y)
    if(level%2):
        if(x == level):
            print((level*level) - (y-1)) 
        
        else:
           print(((level*level) - pow(2,level-1)) + (x-1))
    
    else:
        if(x == level):
            print(((level*level) - pow(2,level-1)) + (y-1))
        
        else:
            print((level*level) - (x-1))

    t-=1
    

